starknet_py/tests/e2e/docs/code_examples/test_gateway_client.py

```diff
@@ -0,0 +1,187 @@
+# pylint: disable=unused-variable
+import pytest
+from starkware.starknet.public.abi import (
+    get_selector_from_name,
+    get_storage_var_address,
+)
+
+from starknet_py.net.client_models import Call
+from starknet_py.net.gateway_client import GatewayClient
+from starknet_py.net.networks import TESTNET
+from starknet_py.tests.e2e.fixtures.constants import MAX_FEE
+
+
+def test_init():
+    # docs-start: init
+    gateway_client = GatewayClient(net=TESTNET)
+    # or (for custom urls)
+    gateway_client = GatewayClient(
+        net={
+            "feeder_gateway_url": "https://alpha4.starknet.io/feeder_gateway",
+            "gateway_url": "https://alpha4.starknet.io/gateway",
+        }
+    )
+    # docs-end: init
+
+
+@pytest.mark.asyncio
+async def test_get_block(gateway_client):
+    # docs-start: get_block
+    block = await gateway_client.get_block(block_number="latest")
+    block = await gateway_client.get_block(block_number=0)
+    # or
+    block = await gateway_client.get_block(block_hash="0x0")
+    # docs-end: get_block
```

Comment:
I think you can also pass "latest" for block_hash.

Reply:
But I'm not sure if it should be included.

Reply:
It shouldn't because if you pass `block_hash=latest` to the gateway URL. It will fail (see https://www.postman.com/starknet-edu/workspace/starknet-edu/request/20082312-e001a8a1-0f05-47d6-a89e-17dc2c6e505b)